### ISSUE TITLE
Do we still need all the representations of sequence lengths in `DataSource`?

### ISSUE THREAD

**Issue by JackKelly** [openclimatefix/nowcasting_dataset#219]
After #213 is implemented, we should check to see if `DataSource` really needs `history_len`, `forecast_len`, `_total_seq_len`, `_total_seq_dur`, `_history_dur` and `_forecast_dur`.  It's possible that some are no longer needed :slightly_smiling_face: 

`_total_seq_len` is used to define `self._shape_of_example` in `NWPDataSource.__post_init__` and `SatelliteDataSource.__post_init__`
